sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/numberFormatExpressions.scala

```diff
@@ -46,8 +46,10 @@ import org.apache.spark.unsafe.types.UTF8String
            grouping separator relevant for the size of the number.
          '$': Specifies the location of the $ currency sign. This character may only be specified
            once.
-         'S': Specifies the position of a '-' or '+' sign (optional, only allowed once).
-         'MI': Specifies that 'expr' has an optional '-' sign, but no '+' (only allowed once).
+         'S': Specifies the position of a '-' or '+' sign (optional, only allowed once at the
+           beginning or end of the format string).
+         'MI': Specifies that 'expr' has an optional '-' sign, but no '+' (only allowed once at the
```

Comment:
Shall we make the doc of `S` and `MI` more consistent? The only difference is `S` allows `+` but `MI` does not.

Reply:
Sure, SG, done.